src/main/java/org/spongepowered/common/mixin/api/data/MixinDataHolder.java

```diff
@@ -65,16 +66,26 @@
     @Override
     public <T extends DataManipulator<?, ?>> Optional<T> get(Class<T> containerClass) {
         TimingsManager.DATA_GROUP_HANDLER.startTimingIfSync();
-        ;
         SpongeTimings.dataGetManipulator.startTimingIfSync();
+        boolean isUser = ((Object) this) instanceof SpongeUser;
+        DataHolder holder = this;
+        if (isUser) {
+            if (((User) this).isOnline()) {
+                holder = ((User) this).getPlayer().get();
```

Comment:
I kind of feel like this should be done in SpongeUser and not put in MixinDataHolder...but would involve duplicate code at that point. Just feels wrong, perhaps either way.

Thoughts @gabizou ?